direct/direct.go

```diff
@@ -335,11 +301,7 @@ func (d *directClient) Call(ctx context.Context, twin uint32, fn string, data in
 		return err
 	}
 
-	d.conM.Lock()
-	if err := d.con.WriteMessage(websocket.BinaryMessage, bytes); err != nil {
-		return err
-	}
-	d.conM.Unlock()
+	d.writer.Write(bytes)
```

Comment:
Maybe because writer i now a channel we can use the context as well otherwise this can get stuck forever if the inner client can't reconnect 

so change this to 
```go
select{
    case d.writer <- bytes:
    case <-ctx.Done():
        return ctx.Err()
}
```